internal/agent/device/applications/podman_monitor.go

```diff
@@ -695,10 +728,20 @@ func (m *PodmanMonitor) updateQuadletContainerStatus(ctx context.Context, app Ap
 	if isFinishedStatus(status) && event.ContainerExitCode == 0 {
 		status = StatusExited
 	}
-	m.updateApplicationStatus(app, event, status, restartCount)
+
+	if status == StatusRemove {
+		if removed := app.RemoveWorkload(event.Name); removed {
+			m.log.Debugf("Removed container: %s", event.Name)
+		}
+		return
+	}
+
+	m.updateApplicationStatus(app, event, "", status, restartCount)
```

Comment:
```suggestion
	m.updateApplicationStatus(app, event, systemdUnit, status, restartCount)
```
is that appropriate in this case?